src/api/app/components/diff_component.html.haml

```diff
@@ -1,7 +1,7 @@
 .pre.scroll
   .diff{ class: "digits-#{diff.max_digits}" }>
-    - diff.lines&.each do |line|
-      - id = "diff_#{file_index}_n#{line.index}"
+    - diff.lines&.[](range)&.each do |line|
```

Comment:
This is getting a bit messy, I would move `diff.lines&.[](range)` into the component class as a method